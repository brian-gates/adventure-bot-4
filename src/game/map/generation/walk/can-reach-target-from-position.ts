import type { Position } from "~/game/map/index.ts";

/**
 * Determine if the target can be reached from the position by moving
 * diagonally.
 */
export function canReachTargetFromPosition({
  target,
  position,
}: {
  target: Position;
  position: Position;
}) {
  const xDistance = Math.abs(position.col - target.col);
  const yDistance = Math.abs(position.row - target.row);
  return xDistance <= yDistance;
}
