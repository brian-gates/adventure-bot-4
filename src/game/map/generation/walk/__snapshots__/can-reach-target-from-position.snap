export const snapshot = {};

snapshot[`isValidPosition to boss 1`] = `
"OOOOOOOOOO
OOOOOOOOOO
OOOOOOOOOO
OOOOOOOOOO
OOOOOOOOOO
OOOOOOOOOO
OOOOOOOOOO
OOOOOOOOOO
OOOOOOOOOO
OOOOOOOOO.
"
`;

snapshot[`isValidPosition to start 1`] = `
"...O......
..OOO.....
.OOOOO....
OOOOOOO...
OOOOOOOO..
OOOOOOOOO.
OOOOOOOOOO
OOOOOOOOOO
OOOOOOOOOO
OOOOOOOOOO
"
`;

snapshot[`isValidPosition: 10x10 1`] = `
"OOOOOOOOOO
.OOOOOOOOO
..OOOOOOO.
...OOOOO..
....OOO...
.....O....
....OOO...
...OOOOO..
..OOOOOOO.
.OOOOOOOOO
"
`;

snapshot[`isValidPosition: 10x10 reverse 1`] = `
"...O......
..OOO.....
.OOOOO....
OOOOOOO...
OOOOOOOO..
OOOOOOOOO.
OOOOOOOOOO
OOOOOOOOOO
OOOOOOOOOO
OOOOOOOOOO
"
`;
